metrics/prometheus_test.go

```diff
@@ -0,0 +1,139 @@
+package metrics
+
+import (
+	"fmt"
+	"github.com/box/kube-applier/run"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"net/http/httptest"
+	"regexp"
+	"testing"
+	"time"
+)
+
+type testCase struct {
+	successes []run.ApplyAttempt
+	failures  []run.ApplyAttempt
+	latency   float64
+}
+
+// TestPrometheusProcessResult tests the processResult() function to ensure that the metrics page is updated properly.
+// With each "test case", we construct a fake run.Result and call processResult.
+// We then make a request to the metrics page handler and parse its raw output.
+// Regular expressions are used to check that the raw output has the correct state for each metric.
+// Note that filenames are reused in order to ensure that the metrics update iteratively.
+func TestPrometheusProcessResult(t *testing.T) {
+	assert := assert.New(t)
+	runMetrics := make(chan run.Result, 5)
```

Comment:
Along those lines. 
I imagine you would need to pass bunch of parameters too. 
And also better names. 

```
TestPrometheusProcessResult() {
    testFullRunWithSuccessOnly(param1, param2, param3)
    testQuckRunMultipleSuccessMultipleFailure(param1, param2, param3 )
    ....
    ...
}
```

What do you think ?